packages/query-core/src/queryClient.ts

```diff
@@ -608,16 +348,21 @@ export class QueryClient {
   }
 
   setQueryDefaults(
-    queryKey: QueryKey,
-    options: QueryObserverOptions<unknown, any, any, any>,
+    options: WithRequired<
+      QueryObserverOptions<unknown, any, any, any>,
+      'queryKey'
+    >,
```

Comment:
also here, not sure if these functions should be changed 🤔 . Would users still know that they'd be setting options for a specific key if we have:

```
queryClient.setQueryDefaults({
  queryKey: ['todos'],
  staleTime: 1000
})
```

?  Maybe it would be better to keep this as two arguments, and instead have the `queryKey` be _omitted_ from the options?

Reply:
Done ✅